lib/ruby_ami/pooled_stream.rb

```diff
@@ -0,0 +1,44 @@
+require "connection_pool"
+
+module RubyAMI
+  class PooledStream
+    def initialize(options)
+      self.options = options
+    end
+
+    def run
+      self.reader_stream = RubyAMI::Stream.new(*option_stream_values)
+      create_writer_pool
+      reader_stream.run
+    end
+
+    def method_missing(method, *args, &block)
+      writer_stream_pool.with do |stream|
+        stream.__send__(method, *args, &block)
```

Comment:
> There is no provision for repairing or checking the health of a connection; connections should be self-repairing. This is true of the Dalli and Redis clients. -- https://github.com/mperham/connection_pool#notes

If the socket owned by `stream` breaks, is it possible that we end up with a tainted connection pool? If so, is there any way for us to heal the broken connections in the pool?

It _might_ be worth having a look at how [`#with` and `#checkin`](https://github.com/mperham/connection_pool/blob/f83b6304c0e5936b1b286b26a73f3febda051c9b/lib/connection_pool.rb#L119-L143) work on the `ConnectionPool` object to see if there's anything we can leverage.